hepcrawl/spiders/common/oaipmh_spider.py

```diff
@@ -70,30 +71,55 @@ def __init__(
         self._crawled_records = {}
 
     def start_requests(self):
+        if self.identifier:
+            return self.start_requests_single(
+                self.url,
+                self.format,
+                self.identifier,
+            )
+        return self.start_requests_sets(
+            self.url,
+            self.format,
+            self.sets,
+            self.from_date,
+            self.until_date,
+        )
```

Comment:
My previous comment was unfortunate... anyhow. This is ok for now, but can you think of different ways to improves this pattern? (we can chat IRL)